src/keycloak/chart/templates/istio-admin.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: keycloak-block-admin-access-from-public-gateway
```

Comment:
Out of caution, I'd strongly recommend a STRICT PeerAuthentication policy in the `keycloak` namespace.

This policy uses negation on namespaces... but requests _don't have_ a namespace if they come over a permissive connection. In practice UDS currently is STRICT by default including keycloak. I think it would be good defensive configuration to have a namespace-local PeerAuthentication (and really strong warnings that you shall not turn this off).

If the mTLS policy is PERMISSIVE, then cluster-internal HTTP requests to http://keycloak-http.keycloak.svc.cluster.local:8080 will basically bypass this DENY AuthorizationPolicy because these requests don't have a namespace set.

See https://istio.io/latest/news/security/istio-security-2021-004/ for further context.

Reply:
Think this makes sense to do @bburky, and we could just force STRICT with no config for it (of course someone could delete but then they more likely know they did something bad).

Because I confused myself, tacking onto this thread that any access from the tenant gateway won't actually hit this DENY policy I don't think. The VS created by `src/keycloak/chart/templates/uds-package.yaml` will already redirect these requests to the `uds` realm (or other configured realm), so this DENY would primarily affect other in-cluster traffic (which I think makes your mention here more important).

Reply:
Yes, if the VirtualService redirect is working correctly, then you'll never hit the DENY AuthorizationPolicy. However, we want to use AuthorizationPolicies for enforcement, don't trust VirtualService configs to actually work. In practice, I don't currently know a way to bypass the VirtualService redirect. I tested this DENY AuthorizationPolicy by temporarily deleting the VirtualService redirect.

Also, this security model is important for Authservice too. There are a few ways to trick Authservice and bypass it. But the AuthorizationPolicy does it's job and denies any requests that don't have an authorized JWT.